src/main/java/seedu/address/model/item/Item.java

```diff
@@ -2,43 +2,85 @@
 
 import java.util.Objects;
 
+import java.time.LocalDateTime;
+import java.time.Period;
+
+import seedu.address.commons.util.CollectionUtil;
 import seedu.address.model.tag.UniqueTagList;
 
 /**
  * Represents a Item in the address book. Guarantees: details are present and
  * not null, field values are validated.
  */
-public abstract class Item implements ReadOnlyItem{
+public class Item {
 
 
     private UniqueTagList tags;
 	private Description description;
 	private boolean isDone;
 	
+	private LocalDateTime startDate;
+    private LocalDateTime endDate;
+	
+	// for recurring events only
+	private boolean isRecurring;
+	private Period recurInterval;
+	private LocalDateTime recurEndDate;
+	
+	
     /**
      * Every field must be present and not null.
      */
 	public Item(Description desc) {
 		// assert !CollectionUtil.isAnyNull(name, phone, email, address, tags);
+	    assert desc != null;
 		this.description = desc;
 		this.isDone = false;
 		// this.tags = new UniqueTagList(tags); // protect internal tags from
 		// changes in the arg list
     }
-
-
+	
+	public Item(Description desc, LocalDateTime start, LocalDateTime end) {
+	    assert !CollectionUtil.isAnyNull(desc, start, end);
+	    this.description = desc;
+	    this.startDate = start;
+	    this.endDate = end;
+	}
+	
+	public Item(Description desc, LocalDateTime end) {
+	    assert !CollectionUtil.isAnyNull(desc, end);
+	    this.description = desc;
+	    this.endDate = end;
+	}
+	
 	public Description getDescription() {
 		return description;
     }
-	
-	public boolean getIsDone() {
-	    return this.isDone;
+
+    public boolean getIsDone() {
+	    return isDone;
 	}
 
-	// @Override
-	// public UniqueTagList getTags() {
-	// return new UniqueTagList(tags);
-	// }
+	
+	public LocalDateTime getStartDate() {
+        return startDate;
```

Comment:
Yes, it should return `null`. All items (events, floating, deadlines) are of type `Item` and are differentiated only by the fields which are filled in. A floating task should return `null` when `getStartDate()` or `getEndDate()` is called because said fields are not defined for a floating task.